### PR TITLE
feat: pagination link component (resolves #10)

### LINKED ISSUE #10: Component: pagination link
Implement pagination links in four colour variations, with all states.

## Additional context or notes

https://www.figma.com/file/LXnq7rdbsr2yyxOG2rSh5S/Inclusive-Design-Guide-2023?type=design&node-id=39-1327&mode=design&t=WibM2Mq2lHqyiZx8-4

### PR DESCRIPTION
**Pull request by chosww**
See #10 for details.

### CODE REVIEW

File: src/static/css/_pagination.css [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+.pagination-previous svg {
+    transform: rotate(180deg);
+}
+
+.pagination-previous:hover {
+    background:
+        linear-gradient(
+            269.79deg,
```
Comment (greatislander) [inline src/static/css/_pagination.css:25]: I think this should be 270deg.

File: src/static/css/_pagination.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            var(
+                --fl-bgColor,
+                var(--pagination-bg)
+            ) -0.44%,
```
Comment (greatislander) [inline src/static/css/_pagination.css:29]: I think this should be 0%.

File: src/static/css/_pagination.css [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            rgb(236 238 255 / 0%) 99.82%
```
Comment (greatislander) [inline src/static/css/_pagination.css:30]: It looks to me like this should be `var(--fl-bgColor, var(--color-white))`.

File: src/static/css/_pagination.css [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        );
+}
+
+.pagination-next:hover {
+    background: linear-gradient(90deg, var(--fl-bgColor, var(--pagination-bg)) 0%, rgb(236 238 255 / 0%) 100%);
```
Comment (greatislander) [inline src/static/css/_pagination.css:35]: Same as above, I think the `rgb` value should be `var(--fl-bgColor, var(--color-white))`.

**Review by greatislander — CHANGES_REQUESTED** [review]
A couple changes to the gradient. Also, this still needs focus and active states!

**Review by greatislander — CHANGES_REQUESTED** [review]
A couple changes to the gradient. Also, this still needs focus and active states!

**Comment by chosww** [comment]
> A couple changes to the gradient. Also, this still needs focus and active states!

Thanks for your review! I got the gradient values from Figma, so I assumed that would be correct. I forgot about the focus and active styles, I will add them now.

**Comment by greatislander** [comment]
> > A couple changes to the gradient. Also, this still needs focus and active states!
> 
> Thanks for your review! I got the gradient values from Figma, so I assumed that would be correct. I forgot about the focus and active styles, I will add them now.

We were both right with respect to the colours— the gradients actually go to `color-blue-100`, not white. I converted the RGB to HSL and it matched. Can you make that change?

**Comment by greatislander** [comment]
One issue with the focus state is that the focus ring as shown in the design should be inside the box and not outside.

You can see an example of how to address this using box-shadow on my draft PR for navigation:

https://github.com/inclusive-design/idg-design-system/blob/feat/navigation/src/static/css/_navigation.css#L19

https://github.com/inclusive-design/idg-design-system/blob/0479bcf5b398a8a2e31ae7d4f9a1967d934f8f15/src/static/css/_navigation.css#L89

Since you won't be using the `outline` for focus styles, you can set its colour to transparent.

**Review by greatislander — CHANGES_REQUESTED** [review]
One other note— even in the default state I think the top border and color should use `--fl-linkColor` to show that it's an interactive element.

**Comment by greatislander** [comment]
> Thanks for your feedback! I set outline to "none" instead, since we don't use it, but if you think setting it to transparent is more appropriate, I will change it to transparent. Also, I added align-items center to the pagination class, because I noticed that the text in the button is not vertically centered.

The reason for the outline rule is described in the link below— and actually you should follow the syntax shown here (`outline: 3px solid transparent;`): https://sarahmhigley.com/writing/whcm-quick-tips/#1.-custom-focus-styles-%2B-a-transparent-outline

**Comment by chosww** [comment]
> > Thanks for your feedback! I set outline to "none" instead, since we don't use it, but if you think setting it to transparent is more appropriate, I will change it to transparent. Also, I added align-items center to the pagination class, because I noticed that the text in the button is not vertically centered.
> 
> The reason for the outline rule is described in the link below— and actually you should follow the syntax shown here (`outline: 3px solid transparent;`): https://sarahmhigley.com/writing/whcm-quick-tips/#1.-custom-focus-styles-%2B-a-transparent-outline

Thanks for your explanation and the link!

File: src/static/css/_pagination.css [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            var(--fl-bgColor, var(--color-blue-100)) 100%
+        );
+}
+
+.pagination:active {
+    --pagination-bg: var(--color-white);
+}
+
+.pagination:focus {
+    box-shadow: 0 0 0 0.2rem var(--fl-linkColor, var(--pagination-box-shadow-color)) inset;
```
Comment (greatislander) [inline src/static/css/_pagination.css:52]: The width is 4px, or 0.25rem (4 / 16).

File: src/static/css/_pagination.css [hunk 6]
```diff
@@ -0,0 +1,73 @@
+.pagination {
+    align-items: center;
+    background: var(--fl-bgColor, var(--pagination-bg));
+    box-shadow: inset 0 0.2rem var(--fl-linkColor, var(--pagination-box-shadow-color));
```
Comment (greatislander) [inline src/static/css/_pagination.css:4]: The width is 4px, or 0.25rem (4 / 16).

File: src/_includes/components/pagination/pagination.njk [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<div class="flex-col gap-m">
+    <div>
+        <button class="pagination pagination-previous {{ params.variantClass }}">
```
Comment (greatislander) [inline src/_includes/components/pagination/pagination.njk:3]: These should be links, not buttons— can't believe I missed that before!

File: src/static/css/_pagination.css [hunk 8]
```diff
@@ -0,0 +1,73 @@
+.pagination {
+    align-items: center;
+    background: var(--fl-bgColor, var(--pagination-bg));
```
Comment (greatislander) [inline src/static/css/_pagination.css:3]: Add a third fallback to `color-blue-100` here so you don't need to define a variable globally for this:

```suggestion
    background: var(--fl-bgColor, var(--pagination-bg, var(--color-blue-100));
```

File: src/static/css/_pagination.css (hunk 6 above)
Comment (greatislander) [inline src/static/css/_pagination.css:4]: Add a third fallback to `--color-blue-600` here so you don't need to define a variable globally for this:

```suggestion
    box-shadow: inset 0 0.2rem var(--fl-linkColor, var(--pagination-box-shadow-color, var(--color-blue-600)));
```

File: src/static/css/_pagination.css [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    box-shadow: inset 0 0.2rem var(--fl-linkColor, var(--pagination-box-shadow-color));
+    color: var(--fl-fgColor, var(--pagination-color));
```
Comment (greatislander) [inline src/static/css/_pagination.css:5]: Add a third fallback to `--color-blue-600` here so you don't need to define a variable globally for this:

```suggestion
    color: var(--fl-fgColor, var(--pagination-color, var(--color-blue-600)));
```

File: src/static/css/_variables.css [hunk 10]
```diff
@@ -72,6 +72,11 @@
         segoe ui symbol
         serif;
 
+    /* Pagination */
+    --pagination-bg: var(--color-blue-100);
+    --pagination-box-shadow-color: var(--color-blue-600);
+    --pagination-color: var(--color-blue-600);
```
Comment (greatislander) [inline src/static/css/_variables.css:78]: These can be removed.

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — CHANGES_REQUESTED** [review]
Final change: the pagination link should not be underlined, so you'll need to add a CSS rule to remove the underline.

**Review by greatislander — APPROVED** [review]

**Pull request merged**